core/opendaq/component/include/opendaq/component.h

```diff
@@ -39,6 +39,12 @@ BEGIN_NAMESPACE_OPENDAQ
  * [includeHeader("<coretypes/event_wrapper.h>")]
  */
 
+enum class ComponentErrorState : EnumType
+{
+    Ok = 0,
+    Warning,
+    Error
+};
```

Comment:
I would not have this on a public header. This should be in the component implementation file.